### ISSUE TITLE
Go to def on `in` should navigate to `__contains__`

### ISSUE THREAD

**Issue by debonte** [microsoft/pylance-release#4467]
2023.6.11 adds the ability to navigate to the implementation of an operator. For example, go to def on `+` takes you to the relevant `__add__` method. It would be good to support the same behavior for keywords such as `in`.

```python
class Bar:
    def __contains__(self, item):
        return item == "foo"

x = Bar()
if "foo" in x: # go to def on in here should navigate to __contains__ on line 2
    print("yes")
else:
    print("no")
```

**Comment by debonte** [comment]
This issue has been fixed in prerelease version 2023.8.51, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023851-30-August-2023-PreRelease)

**Comment by debonte** [comment]
@jrom99 please open a new issue.
